### PR TITLE
Dont modify the merged options when building the confidential client

### PR DESCRIPTION
**Pull request by keegan-caruso**
# Don't mutate options

## Description

PrepareAuthorityInstanceForMsal modifies a shared options object, we shouldn't do that.

### CODE REVIEW

**Comment by keegan-caruso** [comment]
Don't merge yet, need to do more validation

**Review by jennyf19 — APPROVED** [review]
:shipit:

I like it.

**Comment by jennyf19** [comment]
@keegan-caruso is this one ready?

File: src/Microsoft.Identity.Web.TokenAcquisition/MergedOptions.cs [hunk 1]
```diff
@@ -466,14 +474,14 @@ public void PrepareAuthorityInstanceForMsal()
             if (IsB2C && Instance.EndsWith("/tfp/", StringComparison.OrdinalIgnoreCase))
```
Comment (jmprieur) [inline src/Microsoft.Identity.Web.TokenAcquisition/MergedOptions.cs:474]: Maybe a better fix could be to do it:
- here: https://github.com/AzureAD/microsoft-identity-web/blob/455c68776f44ec574738a0fc1b7b1f6039789e44/src/Microsoft.Identity.Web/WebApiExtensions/MicrosoftIdentityWebApiAuthenticationBuilderExtensions.cs#L183
- and here: https://github.com/AzureAD/microsoft-identity-web/blob/455c68776f44ec574738a0fc1b7b1f6039789e44/src/Microsoft.Identity.Web/WebAppExtensions/MicrosoftIdentityWebAppAuthenticationBuilderExtensions.cs#L325

Moving the code in AuthorityHelpers?

**Review by jmprieur — APPROVED** [review]
LGTM but we'd want to:
- test the B2C test apps manually
- run a unified build with this branch

I also proposed an alternative fix

File: src/Microsoft.Identity.Web.TokenAcquisition/MergedOptions.cs (hunk 1 above)
Comment (keegan-caruso) [reply src/Microsoft.Identity.Web.TokenAcquisition/MergedOptions.cs:474]: That should also work.

**Pull request merged**